src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,314 @@
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+from sentry_relay.consts import SPAN_STATUS_CODE_TO_NAME
+from snuba_sdk import Column
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.constants import ObjectStatus
+from sentry.models.project import Project
+from sentry.search.events.builder.spans_indexed import SpansIndexedQueryBuilder
+from sentry.search.events.types import ParamsType
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+
+
+@region_silo_endpoint
+class OrganizationTransactionDetailsEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.PERFORMANCE
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, organization, project_slug, transaction_id):
+        try:
+            project = Project.objects.get(
+                slug=project_slug, organization_id=organization.id, status=ObjectStatus.ACTIVE
+            )
+        except Project.DoesNotExist:
+            return Response(status=404)
```

Comment:
why does this inherit from `OrganizationEndpoint` if a project is always required? why not use `ProjectEndpoint`?

Reply:
Cargo culting from the endpoint I'm replacing, tbh (`OrganizationEventDetailsEndpoint`, which also requires project). If `ProjectEndpoint` does this for me then that makes sense, I'll switch it over.